platform/storage/posix/src/mender-storage.c

```diff
@@ -248,13 +248,23 @@ mender_storage_set_provides(mender_key_value_list_t *provides) {
 }
 
 mender_err_t
-mender_storage_get_provides(mender_key_value_list_t **provides) {
+mender_storage_set_provides(mender_key_value_list_t *provides) {
+    return mender_set_provides(provides, MENDER_STORAGE_NVS_PROVIDES);
+}
+
+mender_err_t
+mender_storage_set_new_provides(mender_key_value_list_t *new_provides) {
+    return mender_set_provides(new_provides, MENDER_STORAGE_NVS_NEW_PROVIDES);
+}
+
+static mender_err_t
+mender_get_provides(mender_key_value_list_t **provides, const char *filename) {
```

Comment:
The difference in name between `mender_storage_get_provides()` and `mender_get_provides()` here can be quite confusing. The `mender_` prefix is not needed here for a `static` function, but maybe the name could reflect it reads provides from a file?